api/nakadi-event-bus-api.yaml

```diff
@@ -606,22 +488,394 @@ paths:
+        not important) - the subscription will be created only once and for all other calls it will just return
+        the subscription that was already created.
+      parameters:
+        - name: subscription
+          in: body
+          description: Subscription to create
+          schema:
+            $ref: '#/definitions/Subscription'
+          required: true
+      responses:
+        '200':
           description: |
-            The flow id of the request, which is written into the logs and passed to called
-            services. Helpful for operational troubleshooting and log analysis.
+            Subscription for such parameters already exists. Returns subscription object that already
+            existed.
+          schema:
+            $ref: '#/definitions/Subscription'
+          headers:
+            Content-Location:
+              description: the URI of existing subscription resource
+              type: string
+            Location:
+              description: the URI of existing subscription resource
+              type: string
+        '201':
+          description: Subscription was successfuly created. Returns subscription object that was created.
+          schema:
+            $ref: '#/definitions/Subscription'
+          headers:
+            Content-Location:
+              description: the URI of created subscription resource
+              type: string
+            Location:
+              description: the URI of created subscription resource
+              type: string
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '403':
+          description: Client is not authorized to perform this operation
+          schema:
+            $ref: '#/definitions/Problem'
+        '400':
+          description: Bad Request
+          schema:
+            $ref: '#/definitions/Problem'
+        '422':
+          description: Unprocessable Entity
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+        '503':
+          description: Service (temporarily) unavailable
+          schema:
+            $ref: '#/definitions/Problem'
+    get:
+      tags:
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: Lists all subscriptions that exist in a system.
+      parameters:
+        - name: owning_application
+          in: query
+          description: parameter to filter subscriptions list by owning application.
           type: string
-          format: flow-id
+          required: false
       responses:
         '200':
           description: OK
           schema:
-            $ref: '#/definitions/Partition'
+            type: object
+            properties:
+              data:
+                type: array
```

Comment:
> But array here is just a field of response object, so we can add anything we want to response object. 

You can extend each item in the array, but you can't say anything about the overall set of partitions. For example an array can't support pagination links (which is why you see things end up in headers because it's the only extension option). Since other result lists will need things like paging (eg subscriptions), the API might as well take a consistent approach overall.